app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -560,6 +560,24 @@ def package_names_for_job_template(action:, search:, versions: nil)
         end
       end
 
+      def deb_names_for_job_template(action:, search:)
```

Comment:
I wonder if we could avoid adding a new method here, and just check `self.operatingsystem.family` within `package_names_for_job_template` instead?

Reply:
@jeremylenz I think, it is a good idea and I have a added a commit on top of the actual commit for the host details page. However, rubocop does not like the changes. Not sure if I can make this any simpler. 

[2024-01-05T09:07:35.947Z] Offenses:

[2024-01-05T09:07:35.947Z] /home/jenkins/workspace/katello-pr-test/app/models/katello/concerns/host_managed_extensions.rb:`541:7: C: Metrics/AbcSize: Assignment Branch Condition size for package_names_for_job_template is too high. [<9, 56, 21> 60.48/60]`

[2024-01-05T09:07:35.947Z]       def package_names_for_job_template(action:, search:, versions: nil) ...

[2024-01-05T09:07:35.947Z]       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2024-01-05T09:07:35.947Z] /home/jenkins/workspace/katello-pr-test/app/models/katello/concerns/host_managed_extensions.rb:`541:7: C: Metrics/CyclomaticComplexity: Cyclomatic complexity for package_names_for_job_template is too high. [16/14]`

[2024-01-05T09:07:35.947Z]       def package_names_for_job_template(action:, search:, versions: nil) ...

[2024-01-05T09:07:35.947Z]       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[2024-01-05T09:07:35.947Z] /home/jenkins/workspace/katello-pr-test/app/models/katello/concerns/host_managed_extensions.rb:`541:7: C: Metrics/MethodLength: Method has too many lines. [34/30]`

[2024-01-05T09:07:35.947Z]       def package_names_for_job_template(action:, search:, versions: nil) ...

Reply:
@nadjaheitmann Two options here. I would be fine with either one:

1. Extract your `if / else` block to its own method. I would probably also do it as a case statement `(case operatingsystem.family`)
2. Or, just disable that cop and enable it at the end of the method. We do that in several places.

Reply:
I tried option one, although I did not want to use a case statement. Seems safer to me to have Debian and everything else :)